Automattic/wp-calypso
# Google My Business: Use feature check instead of plan selector

#### Changes proposed in this Pull Request

* Replaces `isSiteGoogleMyBusinessEligible` selector with feature checks.
* Removes selector `isSiteGoogleMyBusinessEligible`.

#### Testing instructions



* Navigate to https://container-jolly-booth.calypso.live/google-my-business and select a site.
* On a site that is below a Business plan it should redirect to Home.
* Sites with the Google My Business features should show the appropriate screen based on setup state <img width="1510" alt="image" src="https://user-images.githubusercontent.com/1398304/168134033-e00a6e8a-afcf-49f9-a964-064c389744df.png">

* Navigate to https://container-jolly-booth.calypso.live/marketing/connections/ and select a site.
* On a site that is below a Business plan it should not show the Google My Business service
* Sites with the Google My Business features should show the Google My Business service <img width="1510" alt="Screen Shot 2022-05-12 at 10 32 12 AM" src="https://user-images.githubusercontent.com/1398304/168134735-184f4d59-5c44-4109-8f19-ef46eb3249b0.png">



Related to p4TIVU-a66-p2
